app/components/shared/rejection_reasons/rejection_reason_component.rb

```diff
@@ -2,10 +2,17 @@
 # This component class supports the rendering of a single rejection reason predating structured reasons.
 #
 class RejectionReasons::RejectionReasonComponent < ViewComponent::Base
-  attr_reader :application_choice, :reason
+  attr_reader :application_choice
 
   def initialize(application_choice:)
     @application_choice = application_choice
-    @reason = application_choice.rejection_reason
+  end
+
+  def reason
```

Comment:
I think the _no longer accurately named_ `RejectionReasonComponent` should perhaps accept a reason as a constructor arg.

Reply:
I think this is perhaps dual purposing a single purpose component. I'm sorry I didn't spot this in the previous review as I don't think this is the right thing to do.
These sub-components for rejection reasons are to deal with specific data formats ie. The single `rejection_reason` field on application choice, and the 2 different formats from JSON in the `structured_rejection_reasons` field.

I think we should have a distinct component for offer withdrawals, I think they are sufficiently difference from application rejections that they shouldn't be conflated like this.

Reply:
Done! ♻️